calblueprint/meepanyar
# Site Data Refactor to make similar to CustomerData and InventoryData

[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
Customer Data and Inventory Data have a nice redux structure that allows for selector use and entity adapters. This PR refactors siteData so that it has the same structure. This was done now because we will begin implement functionality around objects that live in `SiteData` such as Financial Summaries and Tariff Plans.


### Changelog
1. Made `siteData` structure identical to `inventoryData` and `customerData`
2. `siteData` now has 2 entity adapters per site: 1 for Financial Summaries and 1 for tariff plans
3. Removed all instances of `currentSite` and switched them to use the selector `selectCurrentSite`

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

## How to review
1. Make sure changing sites still work
2. Make sure all pages that used to use `currentSite` are still reachable
  - Home, Customer Profile, Financial Summary (though not used right now), Confirm Payment
2. Pointers on any of the refactored code would be nice as well

[//]: # 'The order in which to review files and what to expect when testing locally'

## Relevant Links

### Online sources

[//]: # 'Optional - copy links to any tutorial or documentation that was useful to you when working on this PR'

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## Next steps
1. Selectors for SiteData to get Tariff Plans based on meter type

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

### Screenshots
Evidence of no more `currentSite`
![image](https://user-images.githubusercontent.com/35501399/114224173-98b88000-9925-11eb-9978-d37b9ac48b7b.png)

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

CC: @julianrkung

[//]: # 'This tags Julian as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
